File: Solutions/1004/src.py
# 一开始用dp 超时。。。
# 法一：滑动窗口

class Solution:
    def longestOnes(self, A: List[int], K: int) -> int:
        cnt = [0] * 2
        i = 0
        j = 0
        ret = 0
        while i < len(A):
            cnt[A[i]] += 1
            i += 1
            while j < len(A) and cnt[0] > K:
                cnt[A[j]] -= 1
                j += 1
            ret = max(ret, i - j) # or ret = max(ret, cnt[0] + cnt[1])
        return ret

File: Solutions/43/src.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        
        
        if(num1[0] == '0' || num2[0] == '0')
            return "0";
        
        
        string ret(num1.size() + num2.size(), '0');
        int carry = 0;
        for(int i = 0; i < num1.size(); i++)
        {
            for(int j = 0; j < num2.size(); j++)
            {
                int index = i + j + 1;
                int temp = (num1[i] - '0') * (num2[j] - '0') + (ret[index] - '0');
                carry = temp / 10;
                ret[index] = (temp % 10) + '0';
                while(carry != 0 && index >= 0)
                {
                    index--;
                    temp = (ret[index] - '0') + carry;
                    carry = temp / 10;
                    ret[index] = (temp % 10) + '0';
                }
            }
        }
        int index = 0;
        while(ret[index] == '0')
            index++;
        return ret.substr(index);
    }
};